volatility3/framework/plugins/linux/lsof.py

```diff
@@ -1,27 +1,27 @@
-# This file is Copyright 2019 Volatility Foundation and licensed under the Volatility Software License 1.0
+# This file is Copyright 2024 Volatility Foundation and licensed under the Volatility Software License 1.0
 # which is available at https://www.volatilityfoundation.org/license/vsl-v1.0
 #
 """A module containing a collection of plugins that produce data typically
 found in Linux's /proc file system."""
-import logging
+import logging, datetime
 from typing import List, Callable
 
-from volatility3.framework import renderers, interfaces, constants
+from volatility3.framework import renderers, interfaces, constants, exceptions
 from volatility3.framework.configuration import requirements
 from volatility3.framework.interfaces import plugins
 from volatility3.framework.objects import utility
 from volatility3.framework.symbols import linux
 from volatility3.plugins.linux import pslist
+from volatility3.plugins import timeliner
 
 vollog = logging.getLogger(__name__)
 
 
-class Lsof(plugins.PluginInterface):
+class Lsof(plugins.PluginInterface, timeliner.TimeLinerInterface):
     """Lists all memory maps for all processes."""
 
     _required_framework_version = (2, 0, 0)
-
-    _version = (1, 1, 0)
+    _version = (2, 0, 0)
```

Comment:
Linux.sockstat uses `lsof` 1.1.0.  Please could you verify that sockscan works with your changes and bump the lsof dependency version for it in this PR please?

Reply:
So yeah, this might not *actually* require a MAJOR version bump, only a MINOR one (so `(1,2,0)`), since the currently exposed API methods stay the same, it's just the `_generator` that changes.  In that case you won't need to update `sockstat`, otherwise you will.  It's your call which option you choose, but generally not bumping a MAJOR version without reason breaks fewer people's plugins...  5;)

Then it looks ready to go in.  5:)